packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -454,59 +354,69 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		storageAddOnsForPlan,
 	} );
 
-	const onCtaClick = useActionCallback( {
+	const priceString = formatCurrency(
+		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
+		currencyCode || 'USD',
+		{
+			stripZeros: true,
+			isSmallestUnit: true,
+		}
+	);
+
+	const { callback, text, postButtonText, status } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		availableForPurchase,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
 		planSlug,
+		planTitle,
+		priceString,
 		cartItemForPlan,
 		selectedStorageAddOn,
 	} );
 
-	// TODO: Unsure about using free plan as a fallback. We should revisit.
-	const onFreeTrialCtaClick = useActionCallback( {
+	const { callback: freeTrialCallback, text: freeTrialText } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		isFreeTrialAction: true,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
+		// TODO: Unsure about using free plan as a fallback. We should revisit.
 		planSlug: freeTrialPlanSlug ?? PLAN_FREE,
+		planTitle,
+		priceString,
 		cartItemForPlan: { product_slug: freeTrialPlanSlug ?? PLAN_FREE },
+		selectedStorageAddOn,
 	} );
 
 	if ( isWpcomEnterpriseGridPlan( planSlug ) ) {
 		return (
-			<PlanButton planSlug={ planSlug } onClick={ onCtaClick }>
+			<PlanButton planSlug={ planSlug } onClick={ callback }>
 				{ translate( 'Learn more' ) }
 			</PlanButton>
 		);
 	}
 
-	const priceString = formatCurrency(
-		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
-		currencyCode || 'USD',
-		{
-			stripZeros: true,
-			isSmallestUnit: true,
-		}
-	);
-
 	if ( isLaunchPage ) {
```

Comment:
This seems largely `redundant` (given other refactors). We already condition on this property in `use-generate-actions` and pass the corresponding callback/text through. So once we have a generic plan-button render method, we can remove (also the prop `isLaunchPage` altogether from the grid - there's no other use).

Just commenting/thinking out loud :-)

Reply:
Totally agree. This was on my todo list as well :D

Reply:
[dcb42ee](https://github.com/Automattic/wp-calypso/pull/89926/commits/dcb42ee2a2775465d798caeb53dc14bcbb10e5cf)